BFAIR/FIA_MS/database_construction.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+import pandas as pd
+from cobra.core.formula import Formula
+
+__version__ = "0.0.1"
+
+
+INVALID_FORMULA_STR = ["(", "Generic", "R", "X"]
+
+
+def print_formula(elements):
+    """
+    The input dictionary, atoms and their amount, is processed to produce
+    the chemical formula as a string
+
+    Parameters
+    ----------
+    elements: dict
+        The elements that form the metabolite and their corresponding amount
+
+    Returns
+    -------
+    formula: str
+        The formula of the metabolite
+    """
+    formula = "".join([f"{k}{int(v)}" for k, v in elements.items()])
+    return formula
+
+
+def zero_charge(metabolite):
+    """
+    The chemical formulas of the input metabolites are modified to not carry
+    any charge
+
+    Parameters
+    ----------
+    metabolite: cobra.Metabolite
+        A metabolite in the cobra format including additional information
+        (formula, charge, elements)
+
+    Returns
+    -------
+    formula: str
+        The formula of the metabolite
+    """
+    formula = Formula(metabolite.formula)
+    if metabolite.charge != 0:
+        if "desulfurated" in metabolite.name:
+            formula.elements["S"] += metabolite.charge / 2
+            formula = Formula(print_formula(formula.elements))
+        else:
+            if "H" in formula.elements and not np.isnan(metabolite.charge):
+                formula.elements["H"] += -metabolite.charge
+                formula = Formula(print_formula(formula.elements))
+    return formula
+
+
+def is_valid(metabolite):
+    """
+    The validity of the input metabolites is checked. It's invalid if it
+    does not have any formula annotated of if the formula includes previously
+    defined invalid symbols
+
+    Parameters
+    ----------
+    metabolite: cobra.Metabolite
+        A metabolite in the cobra format including additional information
+        (formula, charge, elements)
+
+    Returns
+    -------
+    boolean
+        Valid or invalid metabolite
+    """
+    if not metabolite.formula:
+        return False
+    for string in INVALID_FORMULA_STR:
```

Comment:
Instead of a global parameters, I would recommend having `INVALID_FORMULA_STR` as an optional parameter with the defaults given.  I can foresee the need to expand or modify the list depending upon the reconstruction.